### PR TITLE
Feat/modal on dead end

### PR DESCRIPTION
**Pull request by Volubyl**
## Detailed purpose of the PR

The PR covers several dead end cases  : 

**### CASE 1 :** I try to launch the app even if I have no access to it (platform disabled).

--> Tech : we can detect when a platform is disabled if the server returns 403 request header
--> Product : as a user I must see a modal explaining to me that I can ask my administrtor to (re)activate the mobile APP.

The modal looks like this :

Screenshot 2019-05-13 at 17 43 08

**HOW TO TEST IT ?**

In order to test, try to launch the application with a platform that does not have access to the application (eg : TagHeure). 

**### CASE 2 :** I try to launch the app but I have no recommendations, most popular nor other item to display on my dashboard

--> Tech : before desplaying the dashboard we try to fetch in // the most popular and the recommendations. After those API calls, if no cards were found, we should display the modal. The modal also show a refresh button. When triggering the button the last `redux action` will be call once more.

--> Product : as a user I must see a modal explaining that something happend and I can try to refresh the application by triggering a button

The modal looks like this : 

Screenshot 2019-05-13 at 17 59 38

**HOW TO TEST IT ?**

To do so, go grap a test phone, dowload the PR 289 application version on your phone and log on by scanning the QR code available on `error-modal-staging.coorpacademy.com` platform.

Launch the app, you should see the accurate modal with the refresh button. You could click on it and see the application trying to refresh.

WARNING : no content should be shown because no catalog item is provided on the platform.

- [ ] **Breaking changes ?**  
       If checked, what have you broken ?

- [X] **Extra dependency ?**
      If checked, Which extra dependency did you add ? (name, purpose, link ...). 
React Native Modal (https://github.com/react-native-community/react-native-modal) : a great modal handler library

### Testing Strategy

- [ ] Already covered by tests
- [X] Manual testing
- [X] Unit testing

### CODE REVIEW

**Comment by Fausto95** [comment]
First try:
I had the app already installed and connected to a platform, then downloaded you bundled version(overriding mine), opened it and the modal just popped up!
So this is not good at all.
IMG_5317

Second try:
Totally deleted the app, so this mean(AsyncStorage totally clean) then downloaded you bundled version again. And the modal popped up again!

IMG_5318

**Comment by Volubyl** [comment]
> First try:
> I had the app already installed and connected to a platform, then downloaded you bundled version(overriding mine), opened it and the modal just popped up!
> So this is not good at all.
> 
> first try
> 
> Second try:
> Totally deleted the app, so this mean(AsyncStorage totally clean) then downloaded you bundled version again. And the modal popped up again!
> 
> first try



> First try:
> I had the app already installed and connected to a platform, then downloaded you bundled version(overriding mine), opened it and the modal just popped up!
> So this is not good at all.
> 
> first try
> 
> Second try:
> Totally deleted the app, so this mean(AsyncStorage totally clean) then downloaded you bundled version again. And the modal popped up again!
> 
> first try

@Fausto95 is no more relevant since the right PR version was not deployed

File: src/modules/environment.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const __TEST__ = __ENV__ === 'test';
 export const __DEV__ = __ENV__ === 'development';
 
 export const DEV_TOKEN =
-  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoiNWM3NTFhYTY4ZjI3MTMwMDFiMzBlMWIzIiwidXNhZ2UiOiJtb2JpbGUiLCJob3N0IjoiaHR0cHM6Ly9tb2JpbGUtc3RhZ2luZy5jb29ycGFjYWRlbXkuY29tIiwiZ3JhbnRzIjp7Im1vb2MiOnsiZ3JhbnRzIjp7Im1vYmlsZSI6eyJyb2xlcyI6WyJ1c2VyIl19fX19LCJpYXQiOjE1NTExNzg4MTAsImV4cCI6MjAwMDAwMDAwMCwiYXVkIjoibW9iaWxlIiwiaXNzIjoiY29vcnBhY2FkZW15LWp3dCJ9.vC0vxTO8eNk2crp5_nhPgUF7UVqg8K3yFdj6SzHVNIY';
+  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoiNWNkOTlkZDE3MDcwNTQwMDFiMjE1ZDU5IiwiZ3JhbnRzIjp7Im1vb2MiOnsiZ3JhbnRzIjp7ImVycm9yLW1vZGFsIjp7InJvbGVzIjpbImFkbWluIiwidXNlciIsInJoIiwiZ29kbW9kZSIsImNtcyIsInN5c3RlbSJdfX19fSwiaG9zdCI6Imh0dHBzOi8vZXJyb3ItbW9kYWwtc3RhZ2luZy5jb29ycGFjYWRlbXkuY29tIiwidXNhZ2UiOiJtb2JpbGUiLCJpYXQiOjE1NTc4MTkyODYsImV4cCI6MTU4NzgxOTI4NiwiaXNzIjoiY29vcnBhY2FkZW15LWp3dCJ9.BY_OSlkda5IvsULmpllKYYNzLIGWKv0wX_OJEMgCDBY';
```
Comment (Fausto95) [inline src/modules/environment.js]: why are you changing this?

Comment (Volubyl) [reply src/modules/environment.js]: no reason just forgot to remove my token

File: scripts/pull-translations.js [hunk 2]
```diff
@@ -175,7 +183,19 @@ const generate = async (locale: string) => {
...
+      moocErrorTranslations.modal.mobile.refresh_enjoy_learning
+    ),
+    refreshNotWorking: formatTranslation(moocErrorTranslations.modal.mobile.refresh_not_working),
+    askForHelp: formatTranslation(moocErrorTranslations.modal.mobile.ask_for_help),
+    platformHasBeenDisabled: formatTranslation(
+      moocErrorTranslations.modal.mobile.platform_has_been_disabled
+    ),
+    reactivatePlatform: formatTranslation(moocErrorTranslations.modal.mobile.reactivate_platform),
+    iWantIt: formatTranslation(moocErrorTranslations.modal.mobile.i_want_it),
+    refresh: formatTranslation(moocErrorTranslations.modal.mobile.refresh)
```
Comment (VincentCATILLON) [inline scripts/pull-translations.js]: can you please sort the translations added ?

File: yarn.lock [hunk 3]
```diff
@@ -20,9 +20,9 @@
     chokidar "^2.0.3"
 
 "@babel/cli@^7.4.3":
-  version "7.4.3"
-  resolved "https://registry.npmjs.org/@babel/cli/-/cli-7.4.3.tgz#353048551306ff42e5855b788b6ccd9477289774"
-  integrity sha512-cbC5H9iTDV9H7sMxK5rUm18UbdVPNTPqgdzmQAkOUP3YLysgDWLZaysVAfylK49rgTlzL01a6tXyq9rCb3yLhQ==
+  version "7.4.4"
```
Comment (chrisdugne) [inline yarn.lock]: nova-icons has `"@babel/cli" "^7.4.3"`, maybe we must be `exact` there, not to be affected on such PR

File: src/containers/error-listener.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  isVisible: boolean
+};
+class ErrorModal extends React.PureComponent<Props, State> {
+  props: Props;
+
+  handleAssistancePress = () => {
+    Linking.openURL(`mailto:${assistanceEmail}`);
+  };
+
+  handleOnClose = () => {
```
Comment (VincentCATILLON) [inline src/containers/error-listener.js]: looks present

File: src/containers/error-listener.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    if (this.props.errorType === ERROR_TYPE.NO_CONTENT_FOUND) {
+      this.props.signOut();
+    }
+    this.props.hideModal();
+  };
+
+  handleOnPress = () => {
```
Comment (VincentCATILLON) [inline src/containers/error-listener.js]: handleOn looks present

**Pull request merged**